File: app/src/main/java/com/countrydelight/lintruleproject/java_test/find_view_by_id_test/FindViewByIdTest.java
package com.countrydelight.lintruleproject.java_test.find_view_by_id_test;

import android.view.LayoutInflater;
import android.view.View;

import com.countrydelight.lintruleproject.R;
import com.countrydelight.lintruleproject.databinding.LayoutTestIdBinding;

public class FindViewByIdTest {


    // using find view by id
    public void usingFindViewById(View view) {
        view.findViewById(R.id.hello);
    }

    // using view binding
    public void usingViewBinding(LayoutInflater layoutInflater) {
        LayoutTestIdBinding.inflate(layoutInflater);
    }
}

File: app/src/main/java/com/countrydelight/lintruleproject/kotlin_test/FindViewByIdTest.kt
package com.countrydelight.lintruleproject.kotlin_test

import android.view.LayoutInflater
import android.view.View
import android.widget.TextView
import com.countrydelight.lintruleproject.R
import com.countrydelight.lintruleproject.databinding.LayoutTestIdBinding

class FindViewByIdTest {

    // using find view by id
    fun usingFindViewById(view: View) {
        view.findViewById<TextView>(R.id.hello)
    }


    // using view binding
    fun usingViewBinding(layoutInflater: LayoutInflater) {
        LayoutTestIdBinding.inflate(layoutInflater)
    }
}